### ISSUE TITLE
Discussion about standardizing data

### ISSUE THREAD

**Issue by kspieks** [JacksonBurns/astartes#60]
Typical best practices in data science (as well as this [example](https://scikit-learn.org/stable/auto_examples/cluster/plot_dbscan.html#sphx-glr-auto-examples-cluster-plot-dbscan-py) from sklearn) indicate that feature vectors should normally be z-scored before entering the DBSCAN algorithm. This seems sensible and more generally should probably be done before any distance or similarity metric is applied to create clusters for subsequent data splitting. Perhaps a future PR could add the option to z-score the data. However, I'm also fine if we leave this to the user since I believe the goal of this repo is just to create data splits. To make this generalizable beyond cheminformatics, I think we're intentionally offloading the choice of featurization to users who are subject matter experts in their field and probably know more than we do about how to best represent their problem; for example, maybe some features are more important and z-scoring would obfuscate that idk

**Comment by JacksonBurns** [comment]
> Typical best practices in data science (as well as this [example](https://scikit-learn.org/stable/auto_examples/cluster/plot_dbscan.html#sphx-glr-auto-examples-cluster-plot-dbscan-py) from sklearn) indicate that feature vectors should normally be z-scored before entering the DBSCAN algorithm. This seems sensible and more generally should probably be done before any distance or similarity metric is applied to create clusters for subsequent data splitting. Perhaps a future PR could add the option to z-score the data. 

I had a similar thought while working on KMeans -- I vaguely recall from anecdotal stories that the clustering would run faster on normalized data and considered doing so, but ultimately did not. I think this is important for DBSCAN in particular and would make for a nice optional kwarg in train_test_split, but we probably don't want to make it a default behavior. In the event that users are splitting bit vectors with similarity metrics (rather than distances) it wouldn't make sense.

> However, I'm also fine if we leave this to the user since I believe the goal of this repo is just to create data splits. To make this generalizable beyond cheminformatics, I think we're intentionally offloading the choice of featurization to users who are subject matter experts in their field and probably know more than we do about how to best represent their problem; for example, maybe some features are more important and z-scoring would obfuscate that idk

Also agree with this. I think whenever we introduce pre-processing on the data we expose ourselves to errors. Just looking at the _normalize_split_sizes function in main, we have to build a lot of validation around relatively small changes. Optional could be a good way to deal with this, though, and yet I wonder if anyone would use it were we to put in the effort to developer it.

Leaning towards not adding, but not convinced either way - let me know your thoughts on my thoughts.
